src/AspNet.Security.OAuth.Zoho/ZohoAuthenticationHandler.cs

```diff
@@ -0,0 +1,168 @@
+
+        await Events.CreatingTicket(context);
+        return new AuthenticationTicket(context.Principal!, context.Properties, Scheme.Name);
+    }
+
+    protected override async Task<OAuthTokenResponse> ExchangeCodeAsync(OAuthCodeExchangeContext context)
+    {
+        var nameValueCollection = new Dictionary<string, string?>
+        {
+            ["client_id"] = Options.ClientId,
+            ["client_secret"] = Options.ClientSecret,
+            ["code"] = context.Code,
+            ["redirect_uri"] = context.RedirectUri,
+            ["grant_type"] = "authorization_code"
+        };
+
+        if (context.Properties.Items.TryGetValue(OAuthConstants.CodeVerifierKey, out var codeVerifier))
+        {
+            nameValueCollection.Add(OAuthConstants.CodeVerifierKey, codeVerifier!);
+            context.Properties.Items.Remove(OAuthConstants.CodeVerifierKey);
+        }
+
+        var tokenEndpoint = CreateEndpoint(ZohoAuthenticationDefaults.TokenPath);
+        using var request = new HttpRequestMessage(HttpMethod.Post, tokenEndpoint);
+        request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue(MediaTypeNames.Application.Json));
+        request.Content = new FormUrlEncodedContent(nameValueCollection);
+        request.Version = Backchannel.DefaultRequestVersion;
+
+        using var response = await Backchannel.SendAsync(request, Context.RequestAborted);
+        if (!response.IsSuccessStatusCode)
+        {
+            await Log.ExchangeCodeErrorAsync(Logger, response, Context.RequestAborted);
+            return OAuthTokenResponse.Failed(new Exception("An error occurred while retrieving an access token."));
+        }
+
+        var payload = JsonDocument.Parse(await response.Content.ReadAsStringAsync(Context.RequestAborted));
+
+        return OAuthTokenResponse.Success(payload);
+    }
+
+    /// <summary>
+    /// Creates the endpoint for the Zoho API using the location parameter.
+    /// If the location parameter doesn't match any of the supported locations, the default location (US) is used.
+    /// We don't use the accounts-server parameter due to security reasons.
+    /// </summary>
+    /// <param name="path">The request path.</param>
+    /// <returns>The API endpoint for the Zoho API.</returns>
+    private string CreateEndpoint(string path)
+    {
+        var location = Context.Request.Query["location"];
+
+        var domain = location.ToString().ToLowerInvariant() switch
+        {
+            "au" => "https://accounts.zoho.com.au",
+            "ca" => "https://accounts.zohocloud.ca",
+            "eu" => "https://accounts.zoho.eu",
+            "us" => "https://accounts.zoho.com",
+            "in" => "https://accounts.zoho.in",
+            "jp" => "https://accounts.zoho.jp",
+            "sa" => "https://accounts.zoho.sa",
+            "uk" => "https://accounts.zoho.uk",
+            _ => "https://accounts.zoho.com"
+        };
+
+        return $"{domain}{path}";
```

Comment:
Use a `UriBuilder` to do this not string concatenation. Then it will correctly deal with things like if the path does/doesn't start with a `/`.